.github/workflows/xk6-tests/xk6-js-test/jstest.go

```diff
@@ -58,7 +58,7 @@ func (j *JSTest) Foo(arg float64) (bool, error) {
 
 	ctx := j.vu.Context()
 
-	tags := state.Tags.GetCurrentValues().With("foo", "bar")
+	tags := state.Tags.GetCurrentValues().Tags.With("foo", "bar")
```

Comment:
this seems bad .... Maybe just rename `Tags` to `TagsAndMetadata` for now?

I feel like it will be better if they are separate fields in `state` but this might take longer to do and I can see benefits for both, so I am for renaming *and* opening an issue to discuss the internal API and maybe make changes for the next release.

Reply:
One of the big reasons for `TagsAndMetadata` to be packaged together like this was because, in the original PR, we treated them as 2 flavors of tags on the user-facing side, but internally we were saving them in completely different data structures. 

We needed to lock them at the same time and we needed to remove corresponding tags when adding metadata and vice-versa: 
https://github.com/grafana/k6/blob/cd1e21d128251cd0c58c805acb18a62223be0537/metrics/tags.go#L176-L180
https://github.com/grafana/k6/blob/cd1e21d128251cd0c58c805acb18a62223be0537/metrics/tags.go#L185-L186

Now that it's most likely we won't need to have differently-flavored (indexed / non-indexed) tags, a lot of this feels unnecessary :thinking: I guess the common locking and helper methods are still good to have, as well as the internal "accumulator" for the final atlas `Tags` value, and we don't have a lot of time to rewrite this now... :disappointed: But :+1: for renaming `state.Tags` to `state.TagsAndMeta`

Reply:
Considering we are going to open an issue I'm for not renaming now, and including the refactor as the first thing in the next release. @na-- WDYT?

Reply:
:+1: I am also fine with tackling this in next release, with maybe just a TODO in `lib.State` for now

Reply:
Added https://github.com/grafana/k6/issues/2733 for tracking this